### PR TITLE
LocalMap computeIfPresent, compute, merge incorrect behavior

### PR DESCRIPTION
**Pull request by Sammers21**
A solution for https://github.com/vert-x3/issues/issues/313

### CODE REVIEW

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 1]
```diff
@@ -112,7 +112,12 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
       checkType(k);
       checkType(v);
       V output = function.apply(k, v);
-      checkType(output);
+
+      // if output is null then user want to remove a given key. See: https://github.com/vert-x3/issues/issues/313
+      if (output != null) {
+        checkType(output);
```
Comment (Sammers21) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: There is also another approach - to make _null_ admitable for _checktype_

Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: So having null check inside _checkType_

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 2]
```diff
@@ -152,7 +157,12 @@ public void close() {
     return (k, v) -> {
       checkType(k);
       V output = function.apply(k, v);
-      checkType(output);
+
+      // if output is null then user want to remove a given key. See: https://github.com/vert-x3/issues/issues/313
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: no need to add this comment

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 3]
```diff
@@ -169,7 +179,12 @@ public void close() {
     return k -> {
       checkType(k);
       V output = function.apply(k);
-      checkType(output);
+
+      // if output is null then user want to remove a given key. See: https://github.com/vert-x3/issues/issues/313
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: no need to add this comment

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: removed

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 3 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: removed

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 4]
```diff
@@ -112,7 +112,11 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
       checkType(k);
       checkType(v);
       V output = function.apply(k, v);
-      checkType(output);
+
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: remove empty line

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 5]
```diff
@@ -112,7 +112,11 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
...
+      if (output != null) {
+        checkType(output);
+      }
+
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: remove empty line

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 6]
```diff
@@ -152,7 +156,11 @@ public void close() {
     return (k, v) -> {
       checkType(k);
       V output = function.apply(k, v);
-      checkType(output);
+
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: remove empty line

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java [hunk 7]
```diff
@@ -152,7 +156,11 @@ public void close() {
...
+      if (output != null) {
+        checkType(output);
+      }
+
```
Comment (vietj) [inline src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: remove empty line

File: src/test/java/io/vertx/test/core/LocalSharedDataTest.java [hunk 8]
```diff
@@ -35,6 +35,30 @@ public void setUp() throws Exception {
...
+  @Test
+  public void deleteElementOnComputeFunctionReturningNull() {
+    LocalMap<String, String> map = sharedData.getLocalMap("foo");
+
+    // put an initial value
+    map.put("hello", "world");
+
+    // retuning null we should remove the entry
+    map.computeIfPresent("hello", (key, oldValue) -> null);
+    assertNull(map.get("hello"));
```
Comment (vietj) [inline src/test/java/io/vertx/test/core/LocalSharedDataTest.java]: we need instead use `contains` and not `get`

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 4 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: done

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 5 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: done

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 6 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: done

File: src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java (hunk 7 above)
Comment (Sammers21) [reply src/main/java/io/vertx/core/shareddata/impl/LocalMapImpl.java]: done

File: src/test/java/io/vertx/test/core/LocalSharedDataTest.java (hunk 8 above)
Comment (Sammers21) [reply src/test/java/io/vertx/test/core/LocalSharedDataTest.java]: Done

**Pull request merged**